src/io/file_descriptor.cr

```diff
@@ -116,7 +116,15 @@ class IO::FileDescriptor < IO
   def pos : Int64
     check_open
 
-    system_pos - @in_buffer_rem.size
+    in_rem = @in_buffer_rem.size
```

Comment:
IO::Buffered should override pos and add this logic. We probably need unbufferd_pos or something like that.

It's always better to open an issue to discuss what should happen and what's the best way we should tackle it.

Thank you!

Reply:
@asterite How?  

`IO::FileDescriptor` needs to override `pos` to use `system_pos` but overriding replaces the `pos` provided by `IO::Buffered`

Should I require an `unbuffered_pos` method?  Or check one with `respond_to?`?

Something else?

Reply:
The way it's done everywhere else in `IO::Buffered` is:
- define an `abstract unbuffered_xxx` method which types that include `IO::Buffered` must define. In this case it should be `unbuffered_pos`
- define `pos` that calls `unbuffered_pos` and adds the count from the current buffer

Does that sound right?

Reply:
Well yes, but wouldn't that be a breaking change?

Reply:
Oh, that's true... but I guess fixing it just for `IO::FileDescriptor` isn't perfect either. Like, there are a lot of types that include `IO::Buffered` that won't work well.

Maybe we can do `responds_to?(:unbuffered_pos)` as you suggested.